integration-libs/opf/base/components/opf-quick-buy/google-pay/google-pay.service.ts

```diff
@@ -367,6 +367,7 @@ export class OpfGooglePayService {
               this.opfCartHandlerService.loadCartAfterSingleProductTransaction(
                 this.transactionDetails
               );
+              this.deleteAssociatedAddresses();
```

Comment:
it fixes the prob of temp addresses added to user address books when cancelling with googlePay.